tests/scripts/wait-for-docker-services.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# set -x
+
+if [[ $# -ne 1 ]]; then
+  echo "Usage: $0 <Mina Daemon GraphQL port>"
+  exit 1
+fi
+
+START=$(date +%s)
+CURRENT_DIR="$(pwd)"
+
+MINA_DAEMON_GRAPHQL_PORT=${1}
+MINA_DAEMON_GRAPHQL_URL="http://localhost:${MINA_DAEMON_GRAPHQL_PORT}/graphql"
+QUERY='{"query": "{ syncStatus }"}'
+MAX_ATTEMPTS=60
+SLEEP_DURATION=10
+attempt=1
+synced=false
+
+while ! nc -z localhost ${MINA_DAEMON_GRAPHQL_PORT}; do
+  sleep 1
+done
+
+echo ""
+echo "Mina Daemon GraphQL port is ready."
+echo "Waiting for network to sync..."
+echo ""
+
+while [ $attempt -le $MAX_ATTEMPTS ] && [ "$synced" = false ]; do
+  response=$(curl -s -X POST -H "Content-Type: application/json" -d "$QUERY" "$MINA_DAEMON_GRAPHQL_URL")
```

Comment:
Do you think adding more debug info in this script would be helpful if things bug out in CI? Maybe some node information in addition to reporting retry logic? There's a balance here, not sure, wdyt?

Reply:
Good point, thank you. We easily can ask `cURL` to be more verbose when it comes to networking comms. But at the same time, as I mentioned in previous comment, I want to extract common "primitives" into separate NPM helper library and to be platform independent this Bash script will be transpiled into the JS code with perhaps more logging. Speaking of Node information, once Docker Container logging will be re-enabled (see TODOs in Yaml files) for GitHub Actions - you will be able to see regular Mina Daemon logs within the CI job output.
And when it comes to "some node information in addition to reporting retry logic" - do you have some specific data in mind?